RecoLocalFastTime/FTLClusterizer/plugins/MTDCPEESProducer.cc

```diff
@@ -41,7 +42,7 @@ void MTDCPEESProducer::fillDescriptions(edm::ConfigurationDescriptions& descript
 }
 
 std::unique_ptr<MTDClusterParameterEstimator> MTDCPEESProducer::produce(const MTDCPERecord& iRecord) {
-  return std::make_unique<MTDCPEBase>(pset_, iRecord.get(ddToken_));
+  return std::make_unique<MTDCPEFromSiPMTimeBTL>(pset_, iRecord.get(ddToken_));
```

Comment:
@pmeridian @casarsa @fabiocos @parbol I'm still concerned about whether this change makes sense.

What is the purpose of the `MTDCPEBase` class? Will other classes be derived from it? If so, it needs to be managed by a factory or a similar pattern.

Reply:
I guess a better name for ```MTDCPEBase``` may be ```MTDCPEGeneric```.
The actual base exposed to clients is still just the ```MTDClusterParameterEstimator```.

The question is though, if ```MTDCPEBase``` (better called ```MTDCPEGeneric```) should still be available.
This somewhat generically named ```MTDCPEESProducer``` could serve as a "factory" to provide either of the producers; otherwise this one perhaps better be renamed to something more explicitly containing "SiPMTimeBTL" (but then a template would probably be better to still serve another (```MTDCPEGeneric```).
I'm somewhat inclined towards the one configurable option.